specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/jobs.json

```diff
@@ -0,0 +1,539 @@
+            "read"
+          ]
+        },
+        "experimentName": {
+          "description": "The name of the experiment the job belongs to. If not set, the job is placed in the \"Default\" experiment.",
+          "type": "string",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        },
+        "properties": {
+          "description": "The job property dictionary. Properties can be added, but not removed or altered.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "x-ms-mutability": [
+            "create",
+            "read",
+            "update"
+          ]
+        },
+        "tags": {
+          "description": "The job tag dictionary. Tags can be added, removed, and updated.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "x-ms-mutability": [
+            "create",
+            "read",
+            "update"
+          ]
+        },
+        "computeBinding": {
+          "$ref": "#/definitions/ComputeBinding"
+        }
+      }
+    },
+    "CodeJobCreateResourceArmDto": {
+      "description": "Azure Resource Management create resource envelope.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/CodeJob"
+        }
+      }
+    },
+    "MachineLearningServiceError": {
+      "type": "object",
+      "description": "Wrapper for error response to follow ARM guidelines.",
+      "properties": {
+        "error": {
+          "description": "The error response.",
+          "$ref": "#/definitions/ErrorResponse",
+          "readOnly": true
+        }
+      }
+    },
+    "ErrorResponse": {
+      "type": "object",
+      "description": "Error response information.",
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "Error code.",
+          "readOnly": true
+        },
+        "message": {
+          "type": "string",
+          "description": "Error message.",
+          "readOnly": true
+        },
+        "details": {
+          "type": "array",
+          "description": "An array of error detail objects.",
+          "items": {
+            "$ref": "#/definitions/ErrorDetail"
+          },
+          "readOnly": true
+        }
+      }
+    },
+    "ErrorDetail": {
+      "type": "object",
+      "description": "Error detail information.",
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "Error code."
+        },
+        "message": {
+          "type": "string",
+          "description": "Error message."
+        }
+      },
+      "required": [
+        "code",
+        "message"
+      ]
+    },
+    "CodeJobResourceArmDto": {
+      "description": "Azure Resource Manager resource Envelope",
+      "type": "object",
+      "x-ms-azure-resource": true,
+      "properties": {
```

Comment:
For AzureResource recommended model is to derive the common properties from [https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v2/types.json#L45](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v2/types.json#L45)

Reply:
We are setting CodeJob up as a ProxyResource. This should cover all the necessary fields? Unless you mean we need to separate out id, name, and type to a separate definition?

Reply:
I updated it to inherit (allOf) from the common-types ProxyResource

Reply:
When I used the common-type ProxyResource, this broke the lint diff:
```
  '  "Channel": "fatal",\n' +
  '  "Text": "swagger-document/compose - FAILED"\n' +
  '},\n' +
  '{\n' +
  '  "Channel": "fatal",\n' +
  `  "Text": "Error: '$.definitions.Resource.properties.id.description' has incompatible values (---\\n>-\\n  Fully qualified resource Id for the resource. Ex -\\n  /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}\\n, ---\\nSpecifies the resource ID.\\n)."\n` +
  '}\n' +
  "(node:5744) UnhandledPromiseRejectionWarning: Error: '$.definitions.Resource.properties.id.description' has incompatible values (---\n" +
  '>-\n' +
  '  Fully qualified resource Id for the resource. Ex -\n' +
  '  /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}\n' +
  ', ---\n' +
  'Specifies the resource ID.\n' +
```

Let me know if I should use the common-type anyway.

Reply:
This is the recommended model to refer common-type for base types like Proxy/Tracked.